src/flagsView.ts

```diff
@@ -0,0 +1,452 @@
+import * as vscode from 'vscode';
+import { FeatureFlag } from './models';
+import { LaunchDarklyAPI } from './api';
+import { Configuration, getIsTreeviewEnabled } from './configuration';
+import { FlagStore } from './flagStore';
+import * as path from 'path';
+
+export class ldFeatureFlagsProvider implements vscode.TreeDataProvider<FlagValue> {
+	private readonly api: LaunchDarklyAPI;
+	private config: Configuration;
+	private flagStore: FlagStore;
+	private flagValues: Array<FlagValue>;
+	private ctx: vscode.ExtensionContext;
+	private _onDidChangeTreeData: vscode.EventEmitter<FlagValue | undefined> = new vscode.EventEmitter<
+		FlagValue | undefined
+	>();
+	readonly onDidChangeTreeData: vscode.Event<FlagValue | undefined> = this._onDidChangeTreeData.event;
+
+	constructor(api: LaunchDarklyAPI, config: Configuration, flagStore: FlagStore, ctx: vscode.ExtensionContext) {
+		this.api = api;
+		this.config = config;
+		this.ctx = ctx;
+		this.flagStore = flagStore;
+		this.start();
+	}
+
+	refresh(): void {
+		this._onDidChangeTreeData.fire();
+	}
+
+	reload(): void {
+		this.getFlags();
+		this.refresh();
+	}
+
+	getTreeItem(element: FlagValue): vscode.TreeItem {
+		return element;
+	}
+
+	getChildren(element?: FlagValue): Thenable<FlagValue[]> {
+		if (!this.flagValues) {
+			return Promise.resolve([new FlagValue(this.ctx, 'No Flags Found.', vscode.TreeItemCollapsibleState.None)]);
+		}
+
+		if (element) {
+			return Promise.resolve(element.children);
+		} else {
+			return Promise.resolve(
+				this.flagValues.map(function(flag) {
+					return flag;
+				}),
+			);
+		}
+	}
+
+	async getFlags() {
+		const flags = await this.api.getFeatureFlags(this.config.project, this.config.env);
+		let flagValues = [];
+		for (const flag of flags) {
+			let item = this.flagToValues(flag);
+			flagValues.push(item);
+		}
+		this.flagValues = flagValues;
+		this.refresh();
+	}
+
+	async start() {
+		this.ctx.subscriptions.push(
+			vscode.commands.registerCommand('ldFeatureFlags.copyKey', (node: FlagValue) =>
+				vscode.env.clipboard.writeText(node.label.split(':')[1].trim()),
+			),
+			vscode.commands.registerCommand('ldFeatureFlags.openBrowser', (node: FlagValue) =>
+				vscode.env.openExternal(vscode.Uri.parse(node.uri)),
+			),
+			vscode.commands.registerCommand('ldFeatureFlags.refreshEntry', () => this.refresh()),
+			registerTreeviewRefreshCommand(this),
+		);
+
+		// Setup listener for flag changes
+		if (this.flagStore.ldClient === undefined) {
+			setTimeout(() => {
+				console.log(this.flagStore.ldClient);
+				this.flagStore.ldClient.on('update', flags => {
```

Comment:
Looks like this is the first provider that needs to catch feature flags on update, due to the persistent need to refresh the view. My recommendation would be to not expose the ldclient itself and expose a wrapped version of the `on` event listener method on `flagStore`

Reply:
I tried that this morning it looks like the type for `.on` is an LDClient itself. If you have time today or tomorrow can we chat about this. I'm not sure how to proceed.

Reply:
For sure, let's pair on this :)